app/knights/battle_preparation.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+
+class Knight:
+    instances = []
+
+    def __init__(self, name: str,
+                 power: int,
+                 hp: int,
+                 armour: list,
+                 weapon: dict,
+                 potion: dict = None,
+                 protection: int = 0
+                 ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.protection = sum([protect.get("protection")
+                               for protect in self.armour])
+        Knight.instances.append(self)
```

Comment:
It is not redundant here, because in this line we sum protection in the armour dist
and in the method apple_potion we add to protection from the "effect"